README.md

```diff
@@ -803,10 +809,65 @@ Thanks also to
 [@housseindjirdeh](https://github.com/housseindjirdeh),
-[@jakearchibald](https://github.com/jakearchibald), and
+[@jakearchibald](https://github.com/jakearchibald),
+[@matt-buland-sfdc](https://github.com/matt-buland-sfdc), and
 [@slightlyoff](https://github.com/slightlyoff)
 for their help in exploring this space and providing feedback.
 
+## Appendix: types of navigations
+
+The web platform has many ways of initiating a navigation. For the purposes of this API, and in particular the [`navigate` event](#navigation-monitoring-and-interception), the following is intended to be a comprehensive list:
+
+- Users triggering navigations via browser UI, including (but not necessarily limited to):
+  - The URL bar
+  - The back and forward buttons
+  - The reload button
+  - Bookmarks
+- `<a>` and `<area>` elements (both directly by users, and programmatically via `element.click()`)
+- `<form>` elements (both directly by users, and programmatically via `element.submit()`)
+- `<meta http-equiv="refresh">`
+- The `Refresh` HTTP response header
+- The `window.location` setter, the various `location.*` setters, and the `location.replace()`, `location.assign()`, and `location.reload()` methods
+- Calling `window.open(url, nameOfSomeWindow)` will navigate a window whose `window.name` is `nameOfSomeWindow`
+- `history.back()`, `history.forward()`, and `history.go()`
+- `history.pushState()` and `history.replaceState()`
+- `appHistory.back()`, `appHistory.forward()`, `appHistory.navigateTo()`
+- `appHistory.pushNewEntry()` and `appHistory.updateCurrentEntry()`
+- [`document.open()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/open)
+
+**Cross-document navigations** are navigations where, after the navigation completes, you end up in a different `Document` object than the one you are curently on. Notably, these unload the old document, and stop running any JavaScript code from there.
+
+**Same-document** navigations are ones where, after the navigation completes, you stay on the same `Document`, with the same JavaScript environment.
+
+Most navigations are cross-document navigations. Same-document navigations can happen due to:
+
+- Any of the above navigation mechanisms only updating the URL's fragment, e.g. `location.hash = "foo"` or clicking on `<a href="#bar">` or calling `history.back()` after either of those two actions
+- `history.pushState()` and `history.replaceState()`
+- `appHistory.pushNewEntry()` and `appHistory.updateCurrentEntry()`
+- `document.open()`
+- [Intercepting a cross-document navigation](#navigation-monitoring-and-interception) using the `appHistory` object's `navigate` event, and calling `event.respondWith()`
+
+Here's a summary table:
+
+|Trigger|Cross- vs. same-document|Fires `navigate`?|`event.userInitiated`|
+|-------|------------------------|-----------------|---------------------|
+|Browser UI (fragment change only)|Always same|Yes|Yes|
+|Browser UI (other)|Always cross|No|—|
```

Comment:
Does this include browser back? For entries that were previously pushed, browser back should be equivalent to `history.back()`. So long as the SPA is pushing correctly, back & forward inside the app's history entries should fire `navigate`. (until the user goes back/forward out of the app's history space).

Reply:
That doesn't work though, because if you allow intercepting backs, even same-app-history ones, then the page can just trap the user forever on their origin, never letting them actually get back to where they were.

If you want to _see_ backs, you can use `currententrychange`. But since we cannot allow them to be interceptable for anti-abuse reasons, we don't fire `navigate` for them.

Reply:
Is it worth adding another column for `currententrychange`? I don't want this nuance to be lost

Reply:
I should try to incorporate it somehow (beyond what's already mentioned in the main section: "These restrictions are designed so that canceling..."). But a dedicated column isn't great, because its values are basically the same as the same- vs. cross-document column. That is, cross-document navigations don't fire `currententrychange` (since by the time the entry has changed, the document is gone, and no event listeners fire on the old document). And same-document navigations always do.

Reply:
Yeah, the back/capture part is clear in the other section, I just rabbit-holed on what this table says on its own.

I think https://github.com/WICG/app-history#current-entry-change-monitoring is also descriptive enough to understand when I'd expect `currententrychange`, even if it's not listed in table format. But just to make sure I do understand... If I do a cross-document navigation from A.foo/ to B.foo/ via a URL-bar edit, then the A document does not hear `navigate` nor `currententrychange`, but the B document does hear a `currententrychange`. Is that correct?

Reply:
This discussion made me realize that we can't intercept the back button even for fragment changes, so I pushed some more changes in that direction...

Reply:
yes, that's a good clarification. I had assumed you were only describing editing the fragment in the URL-bar (a forward fragment change)

Reply:
> But just to make sure I do understand... If I do a cross-document navigation from A.foo/ to B.foo/ via a URL-bar edit, then the A document does not hear navigate nor currententrychange,

Correct

> but the B document does hear a currententrychange. Is that correct?

Well... yes, as currently written, but I'm not sure that's a good idea. See the TODO in https://github.com/WICG/app-history/pull/27. In particular, this would be the only time `currententrychange` fires for a cross-document navigation. And it'd be basically redundant with the `load` event. It's a weird special case, and I suspect firing it might hurt people more than it helps. Probably this should get its own dedicated issue...

Reply:
Started an issue thread: #30 .